### PR TITLE
[site] Fix invalid HTML and enable HTML checking

### PR DESCRIPTION
**Pull request by dshevtsov**
## Purpose of this pull request

This pull request fixes invalid HTML across the site and enables HTML validation in the link checker config. Most of the failures are caused by duplication ID (when the ID is not unique within a page) and closing tags with no opening tags.

### CODE REVIEW

File: guides/v2.2/config-guide/multi-master/multi-master_manual.md [hunk 1]
```diff
@@ -258,7 +258,7 @@ mysqldump -u <your database root username> -p <your main magento DB name> sequen
 
 This script restores sales data in your quote database.
 
-#### NDB requirement
+#### NDB requirement {#sql-sales-restore-ndb-req}
```
Comment (jeff-matthews) [inline guides/v2.2/config-guide/multi-master/multi-master_manual.md]: Why is this necessary?

File: guides/v2.2/install-gde/install/prepare-install.md [hunk 2]
```diff
@@ -59,7 +55,7 @@ If so, create [`auth.json`]({{ page.baseurl }}/install-gde/prereq/dev_install.ht
 
 {% include install/file-system-perms-before_22.md %}
 
-### Next step
+### Next step {#next-step-section}
```
Comment (jeff-matthews) [inline guides/v2.2/install-gde/install/prepare-install.md]: Why is this necessary?

File: guides/v2.2/cloud/before/before-workspace-magento-prereqs.md [hunk 3]
```diff
@@ -113,7 +113,7 @@ These example instructions detail how to install and create a MariaDB database f
 		use magento;
 6. Exit when done.
 
-### Pro: Set up the auto-increment for MariaDB {#cloud-mysql}
+### Pro: Set up the auto-increment for MariaDB
```
[the remark this answers is not in the document]
Comment (dshevtsov) [reply guides/v2.2/cloud/before/before-workspace-magento-prereqs.md]: Yes, it's been used in the same topic that caused an ID duplication error.

File: guides/v2.2/config-guide/multi-master/multi-master_manual.md (hunk 1 above)
Comment (dshevtsov) [reply guides/v2.2/config-guide/multi-master/multi-master_manual.md]: This topic has sections with the same name. The sections with the same name have the same ID by default, that causes a duplication ID error.

File: guides/v2.2/install-gde/install/prepare-install.md (hunk 2 above)
Comment (dshevtsov) [reply guides/v2.2/install-gde/install/prepare-install.md]: The section has several `### Next step` sections. Each of them by default is assigned with the same `id="next-step"`. Each [ID have to be unique within a page](https://www.w3schools.com/html/html_id.asp).

**Comment by dshevtsov** [comment]
running tests
